design/Proposal-to-enhance-schema-for-metal3.md

```diff
@@ -0,0 +1,228 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Proposal to enhance Schema for Metal3
+
+## Table of Contents
+
+<!--ts-->
+   * [Proposal to enhance Schema for Metal3](#title)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+      * [References](#references)
+<!--te-->
+
+## Summary
+
+Enhancing the `metal3` schema by adding useful parameters that we received 
+from ironic introspection which will be helpful in performance based use cases.
+
+## Motivation
+
+To make the existing `metal3` schema compatible and scalable for heavy computation 
+based use cases.  
+
+### Goals
+
+To compare the introspection data fetched from `Ironic-Inspector` through 
+`Gopher-cloud client` library with `metal3` schema and add useful parameters 
+in `metal3` schema.
+
+## Proposal
+
+The comparison of hardware introspection data coming from `Ironic-Inspector` 
+and existing `metal3` schema for configuration of `baremetalhost` concludes 
+that few hardware parameters are missing in `metal3` schema that are fetched 
+from `Ironic-Inspector`. We propose to add these parameters in `metal3` schema for 
+scalability and performance enhancement.
+
+### Implementation Details/Notes/Constraints
+
+Link for Existing `Metal3` Specs
+
+Please refer `metal3` spec for bare-metal:
+https://github.com/metal3-io/baremetal-operator/blob/master/deploy/crds/metal3.io_baremetalhosts_crd.yaml
+
+The investigation of the existing schema indicates that BareMetal-Operator 
+gets BareMetal node introspection data using an API call to `Ironic-Inspector` 
+through `gopher-cloud client` library. 
+
+Having fetched BareMetal Node introspection data from Ironic-Inspector and
+compared it with BareMetalHost schema of `metal3` `(metal3.io_baremetalhosts_crd.yaml`), 
+we recommend following parameters need to be added in schema of `metal3` for `NFV` 
+workloads.
+1. Add enhancement in the existing `metal3` schema for the following parameters:
+
+    CPU Parameters:
+    
+    Parameters for `NUMA` topology:
+     
+    For better performance, selection of CPUs based on the `NUMA` topology 
+    becomes necessary. In case of nodes with `DPDK` aware NICs, the CPUs for 
```

Comment:
This information seems useful for scheduling workloads, but I don't understand how it would be used during provisioning. Can you elaborate on that?

Reply:
For provisioning, this is not useful. We are adding this for future workload scenarios where user can look at the CR and plan there workloads accordingly.

Reply:
https://kcsna2019.sched.com/event/Vv0W/present-and-future-of-hardware-topology-awareness-in-kubelet-connor-doyle is relevant for the scheduler using NUMA topology. I wouldn't want to reproduce data in the BareMetalHost that is going to be available somewhere else, like the node.